### PR TITLE
chore: updates for latest microcluster changes [WD-13691]

### PR DESCRIPTION
**Pull request by mas-who**
## Done
- remove `microcluster.UpdateServer` and use `microcluster.LocalClient.UpdateServer`
- adjusted input options for `lxd.shared.KeyPairAndCA`
- adjusted for `microcluster.Args` inputs and setup `microclusterDaemonArgs` for starting the daemon
- add expiry option for member join token CLI command

### CODE REVIEW

**Review by edlerd — APPROVED** [review]
LGTM, nice cleanups through the upstream changes.

File: cmd/lxd-cluster-mgr/tokens.go [hunk 1]
```diff
@@ -49,6 +50,8 @@ func (c *cmdTokensAdd) command() *cobra.Command {
 		RunE:  c.run,
 	}
 
+	cmd.Flags().DurationVar(&c.flagExpiry, "expiry", 0, "Specify the duration (i.e. 5m or 48h) for the token to be valid")
```
Comment (edlerd) [inline cmd/lxd-cluster-mgr/tokens.go:53]: Do we have a test for this or should we extend/ add one?

Comment (mas-who) [reply cmd/lxd-cluster-mgr/tokens.go:53]: we don't have any tests for cli commands yet, I can add some tests for cli in a separate issue?

Comment (edlerd) [reply cmd/lxd-cluster-mgr/tokens.go:53]: sounds good to me.

**Review by roosterfish — APPROVED** [review]
LGTM!

Before merging this you might want to to wait for the LTS version release so you can track an actual version of `microcluster` and not just a commit. 
But ofc this can also be changed afterwards :)

**Comment by mas-who** [comment]
> LGTM!
> 
> Before merging this you might want to to wait for the LTS version release so you can track an actual version of `microcluster` and not just a commit. But ofc this can also be changed afterwards :)

I think let's merge this so other PRs can also benefit from the latest changes. Will update to LTS version tracking when it's ready

**Pull request merged**